### ISSUE TITLE
problem in font render

### ISSUE THREAD

**Issue by aliamini87** [Phantom1003/obsidian-slide-note#4]
This is the view if the slide from the pdf viewer :

![image](https://github.com/Phantom1003/obsidian-slide-note/assets/91720545/760744ab-e388-45dc-be07-ce9dd231d237)

and this is what the plugin renders:
![image](https://github.com/Phantom1003/obsidian-slide-note/assets/91720545/22f110ef-4e4a-4d35-bed6-0c66f6d275d6)

It seems that there is a problem in font rendering.

**Comment by Phantom1003** [comment]
Can you send this pdf as an attachment to my email (phantom@zju.edu.cn)

**Comment by aliamini87** [comment]
yes

**Comment by Phantom1003** [comment]
The pdf is rendering properly on my machine, maybe some system environment differences are causing this bug.
Please check if you have the relevant fonts installed on your local machine.
Since slide-note uses pdfjs to render the pdf, sorry I can't provide direct advice to help you solve this problem.

**Comment by aliamini87** [comment]
I install the necessary font files and check it by firefox which use pdfjs. The document loaded properly in firefox but nothing changed in the plugin!

**Comment by aliamini87** [comment]
Another note.
I changed the font of some texts in the document by adobe acrobat pro to very common fonts. The font changed but irregularity persisted. I mean that this is not the font problem. I don't know what is this. I only can describe this as irregularity in words arrangement.

**Comment by Phantom1003** [comment]
You can use `dpi` option to increase the resolution

**Comment by aliamini87** [comment]
The resolution is not the problem. words are irregular:

![image](https://github.com/Phantom1003/obsidian-slide-note/assets/91720545/3643e727-83a1-4709-be9c-a9a0ab02f3f8)

instead of

![image](https://github.com/Phantom1003/obsidian-slide-note/assets/91720545/401d9cba-211f-4147-99cd-0f19b3438be1)

**Comment by Phantom1003** [comment]
I upgrade the version of the pdfjs to the latest version, can you manually install this [preview plugin](https://github.com/Phantom1003/obsidian-slide-note/releases/tag/pre-0.0.7)?

**Comment by aliamini87** [comment]
Off course.
But I don't know exactly how should I do that.

**Comment by Phantom1003** [comment]
Please follow [this](https://forum.obsidian.md/t/plugins-mini-faq/7737) guide

**Comment by aliamini87** [comment]
[SlideNote] Render Error: UnknownErrorException: The API version "3.6.172" does not match the Worker version "2.12.313".

**Comment by Phantom1003** [comment]
Slide Note has conflicts with the better-pdf plugin.
By the way, does better-pdf able to render the text font?

**Comment by aliamini87** [comment]
No. 
same problem.

**Comment by aliamini87** [comment]
I disabled better-pdf
new error :
[SlideNote] Render Error: InvalidPDFException: The PDF file is empty, i.e. its size is zero bytes.

**Comment by aliamini87** [comment]
Hi
I've found the problem.
The problem is from RTL option of obsidian.
I created a clean vault with only slide-note installed. before enabling RTL support in obsidian settings, everything was fine. When I enable RTL setting (My language is a RTL language), disordered words were appeared in the render of the plugin.

**Comment by Phantom1003** [comment]
Nice catch, I can replay this bug now.
I will try to fix it, thanks for your report.

**Comment by Phantom1003** [comment]
Can you provide any PDFs written in your RTL language?
I plan to add an option for users to specify their preferred language direction.

**Comment by aliamini87** [comment]
I will send for you.

**Comment by Phantom1003** [comment]
version 0.0.8 should fix this problem

**Comment by Phantom1003** [comment]
The bug was caused by the pdfjs. It will try to access the direction CSS property of the canvas html container, and decide its internal operations.
When you enable the RTL option in obsidian, obsidian will change the direction CSS property to RTL, and that's why we meet this problem.
During my test, I found when pdfjs treat the document as an LTR file, it can always render correctly even if the PDF has both RTL and LTR in it.
If you meet new bugs, welcome to open an issue.
